### ISSUE TITLE
Font weight light (anything < 400) does not render correctly in Flutter Web

### ISSUE THREAD

**Issue by julianctni** [flutter/flutter#75832]
### Steps to reproduce
1. Create a Flutter project
2. Add a Text Widget with font weight < 400 (e.g. w300)
3. Compile it for Flutter Web and view it in Chrome

### Actual Behaviour
The Text Widget is rendered with font weight normal

### Expected Behaviour
The Text Widget is rendered with a light font weight (< 400).

When running the same application on Android, the font weight is rendered correctly.

```
❯ flutter --version
Flutter 1.27.0-2.0.pre.43 • channel master • https://github.com/flutter/flutter.git
Framework • revision 16af4aa7eb (25 hours ago) • 2021-02-10 17:01:04 +0800
Engine • revision b3a18f5182
Tools • Dart 2.13.0 (build 2.13.0-15.0.dev)
```

Thanks in advance!
Best regards

Julian

**Comment by darshankawar** [comment]
@julianctni 
Thanks for reporting this. I think this is related to canvaskit, since using `html` renderer, the `fontWeight` is rendered properly.

```
body: Center(
        child: Text('This is a test', style: TextStyle(
          fontWeight: FontWeight.w300,
        ),
        ),
      )
```

1. With `flutter run -d chrome` which if I am not wrong defaults to canvaskit on desktop browsers shows below:

<img width="249" alt="Screenshot 2021-02-11 at 6 35 41 PM" src="https://user-images.githubusercontent.com/67046386/107641927-3aab4b80-6c9a-11eb-85a0-ca6ce4b566bb.png">

2. With `flutter run -d chrome --web-renderer html` , font weight is rendered properly as:

<img width="191" alt="Screenshot 2021-02-11 at 6 37 36 PM" src="https://user-images.githubusercontent.com/67046386/107642046-5dd5fb00-6c9a-11eb-8e69-1cee1a481360.png">

To confirm, an issue with canvaskit, running `flutter run -d chrome --web-renderer canvaskit` also results in 1 as above.

3. Android shows properly:

<img width="191" alt="Screenshot 2021-02-11 at 6 37 36 PM" src="https://user-images.githubusercontent.com/67046386/107642214-937ae400-6c9a-11eb-9d95-ce1f5ea24df8.png">

There was similar [issue](https://github.com/flutter/flutter/issues/60125) fixed and [merged](https://github.com/flutter/engine/pull/20272) and may have come back in latest master.

Labeling it as regression, since it was verified and confirmed to be working on master 1.21 per this comment, https://github.com/flutter/flutter/issues/60125#issuecomment-673923963

 <details>
<summary>master flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.27.0-2.0.pre.65, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.27.0-2.0.pre.65 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision a8075deea3 (9 hours ago), 2021-02-10 22:46:02 -0500
    • Engine revision cdc49c575b
    • Dart version 2.13.0 (build 2.13.0-25.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • SM G975F (mobile)           • RZ8M802WY0X   • android-arm64  • Android 10
      (API 29)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11
      (API 30) (emulator)
    • macOS (desktop)             • macos         • darwin-x64     • Mac OS X
      10.15.4 19E2269 darwin-x64
    • Chrome (web)                • chrome        • web-javascript • Google
      Chrome 88.0.4324.150

! Doctor found issues in 1 category.

```

</details>

**Comment by yjbanov** [comment]
Might be related to https://github.com/flutter/flutter/issues/74338

**Comment by iapicca** [comment]
this issue seems to keep coming back https://github.com/flutter/flutter/issues/60125
I think would be great if before closing the issue would be added a test to prevent this from happening again

**Comment by yjbanov** [comment]
Hypothesis: the issue seems to be that Skia can turn a regular font to bold (which is why https://github.com/flutter/flutter/issues/60125 is no longer an issue), but it cannot make it thin. When the app doesn't supply fonts we use fallback fonts, and we don't currently have thin fonts in our fallback list. We'll need to add them.

**Comment by sima-11** [comment]
Any update on this one?

**Comment by smallTrogdor** [comment]
Hi,
same as @sima-11 : is there any update on the rendering using CanvasKit with respect to FontWeights? w400 seems to be included in this issue.

```dart
Text(
  'The quick brown fox died.',
  style: TextStyle(
    fontSize: 15.0,
    height: 23.0 / 15.0,
    fontStyle: FontStyle.normal,
    fontWeight: FontWeight.w400,
    fontFamily: 'Roboto',
    color: Colors.black,
  ),
)
```

with `flutter run -d chrome --web-renderer html`

![Screenshot 2022-04-01 at 14 23 37](https://user-images.githubusercontent.com/30844036/161262631-e4ac5553-04c6-4dae-b862-d2a5378102e4.png)

with `flutter run -d chrome`

![Screenshot 2022-04-01 at 14 09 58](https://user-images.githubusercontent.com/30844036/161262744-9e947ad6-b9f6-47c6-8601-e417e45ceceb.png)



<details>
  <summary>flutter doctor</summary>
  [✓] Flutter (Channel stable, 2.10.3, on macOS 12.2.1 21D62 darwin-x64, locale en-GB)
    • Flutter version 2.10.3 at /Users/nico/fvm/versions/2.10.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (4 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/nico/Library/Developer/android
    • Platform android-31, build-tools 30.0.3
    • ANDROID_HOME = /Users/nico/Library/Developer/android
    • Java binary at: /usr/local/Cellar/openjdk@11/11.0.12/libexec/openjdk.jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Homebrew (build 11.0.12+0)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.65.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.36.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 99.0.4844.84

[✓] HTTP Host Availability
    • All required HTTP hosts are available
</details>

**Comment by michaelparkadze** [comment]
Any updates regarding this matter? I've encountered this issue in the latest flutter version. (2.10.5)

**Comment by huycozy** [comment]
I'm able to reproduce this issue on `stable` and `master` channel:

| HTML | Canvaskit |
| --------------- | --------------- |
<img width="360" alt="Screen Shot 2022-04-28 at 17 13 56" src="https://user-images.githubusercontent.com/104349824/165730802-a54cbded-942d-4634-9ffd-c9259d87efaf.png"> | <img width="360" alt="canvaskit" src="https://user-images.githubusercontent.com/104349824/165730515-ed638a27-78bf-4850-a46d-889da9cc1e69.png">

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              '1. The quick brown fox died. - normal',
              style: TextStyle(
                fontSize: 15.0,
                height: 23.0 / 15.0,
                fontStyle: FontStyle.normal,
                fontWeight: FontWeight.normal,
                fontFamily: 'Roboto',
                color: Colors.black,
              ),
            ),
            Text(
              '2. The quick brown fox died. - w300',
              style: TextStyle(
                fontSize: 15.0,
                height: 23.0 / 15.0,
                fontStyle: FontStyle.normal,
                fontWeight: FontWeight.w200,
                fontFamily: 'Roboto',
                color: Colors.black,
              ),
            )
          ],
        ),
      ),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 2.10.5, on macOS 12.2.1 21D62 darwin-x64, locale en-VN)
    • Flutter version 2.10.5 at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5464c5bac7 (10 days ago), 2022-04-18 09:55:37 -0700
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Android Studio
    • Android Studio at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/203.7185775/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 100.0.4896.127

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[✓] Flutter (Channel master, 2.13.0-0.0.pre.793, on macOS 12.2.1 21D62 darwin-x64, locale en-VN)
    • Flutter version 2.13.0-0.0.pre.793 at /Users/huynq/Documents/GitHub/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8fc22a7df2 (44 minutes ago), 2022-04-28 02:44:04 -0400
    • Engine revision 67940eb620
    • Dart version 2.18.0 (build 2.18.0-66.0.dev)
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7042882/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[!] Android Studio
    • Android Studio at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/203.7185775/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 100.0.4896.127

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

**Comment by gcostaapps** [comment]
Any updates? Facing the same issue here

**Comment by DmitrySikorsky** [comment]
I have the same problem in 3.3.1

**Comment by DmitrySikorsky** [comment]
So, I can confirm that at least in 3.3.1 it is easy to fix.

If I have the custom fonts added like

```
  fonts:
      - family: Montserrat
        fonts:
          - asset: fonts/Montserrat_Regular.ttf
          - asset: fonts/Montserrat_Medium.ttf
          - asset: fonts/Montserrat_SemiBold.ttf
          - asset: fonts/Montserrat_Bold.ttf
```

text will be rendered correctly everywhere (desktop Chrome, Windows, Android, iOS) except Chrome on mobile devices. On mobile Chrome all text will have the same (bold) weight.

If I specify font wight explicitly like

```
  fonts:
      - family: Montserrat
        fonts:
          - asset: fonts/Montserrat_Regular.ttf
            weight: 400
          - asset: fonts/Montserrat_Medium.ttf
            weight: 500
          - asset: fonts/Montserrat_SemiBold.ttf
            weight: 600
          - asset: fonts/Montserrat_Bold.ttf
            weight: 700
```

Then font weight on mobile Chrome is fixed too.

**Comment by teneon** [comment]
I have the same problem with Flutter web. w100 to w500 are displayed as regular. And w600 to w900 are displayed as bold, there is nothing in between.

I also tried starting is as suggested:
flutter run -d chrome --web-renderer canvaskit
flutter run -d chrome --web-renderer html

But it doesn't work at all, it's all the same. I am using latest version on Flutter beta channel and i am on running Linux ,browser is Chrome. Everything works as expected on Android emulator though ,so it is just web related.

Hopefully this gets fixed as soon as possible. I am not sure why this issue is closed though?

best regards,
Jim

**Comment by icecandy** [comment]
I can confirm that this is still a problem with rendering smaller font weights in flutter web. I am trying to render a font with weight 300, but it is appearing the same as weight 500.
If I run from Android Studio it does work (with html or canvaskit renderers).
If I build the project with:
`$ flutter build web --web-renderer html`
and upload to my web server it DOES NOT work
If I build the project with:
`$ flutter build web --web-renderer canvaskit`
and upload to my web server it DOES work

So the problem is that the html rendered site does not work.
However, following a suggestion from somewhere else, if I create a "dummy" extra font spec in pubspec.yaml just for the weight 300 font (and also matching this in the web > assets > FontManifest.json file) it does seem to work .
So changing my pubspec.yaml font spec from:

```
fonts:
    - family: Roboto
      fonts:
        - asset: fonts/Roboto-Light.ttf
          weight: 300
        - asset: fonts/Roboto-Medium.ttf
          weight: 500
```


to

```
fonts:
    - family: Roboto
      fonts:
        - asset: fonts/Roboto-Medium.ttf
          weight: 500

    - family: RobotoLight
      fonts:
        - asset: fonts/Roboto-Light.ttf
          weight: 300
```
seems to work.

**Comment by Justus-M** [comment]
Still having this issue, and it has a very negative impact on the design

**Comment by Justus-M** [comment]
> I can confirm that this is still a problem with rendering smaller font weights in flutter web. I am trying to render a font with weight 300, but it is appearing the same as weight 500. If I run from Android Studio it does work (with html or canvaskit renderers). If I build the project with: `$ flutter build web --web-renderer html` and upload to my web server it DOES NOT work If I build the project with: `$ flutter build web --web-renderer canvaskit` and upload to my web server it DOES work
> 
> So the problem is that the html rendered site does not work. However, following a suggestion from somewhere else, if I create a "dummy" extra font spec in pubspec.yaml just for the weight 300 font (and also matching this in the web > assets > FontManifest.json file) it does seem to work . So changing my pubspec.yaml font spec from:
> 
> ```
> fonts:
>     - family: Roboto
>       fonts:
>         - asset: fonts/Roboto-Light.ttf
>           weight: 300
>         - asset: fonts/Roboto-Medium.ttf
>           weight: 500
> ```
> 
> to
> 
> ```
> fonts:
>     - family: Roboto
>       fonts:
>         - asset: fonts/Roboto-Medium.ttf
>           weight: 500
> 
>     - family: RobotoLight
>       fonts:
>         - asset: fonts/Roboto-Light.ttf
>           weight: 300
> ```
> 
> seems to work. Flutter 3.10.5 stable.

Thanks for sharing this. Could you please also show the json file and your flutter code?

I wasn't able to make it work.

Here's my flutter code:

```
Text(
            "Easy login",
            style: TextStyle(
                fontSize: 32,
                fontFamily: 'RobotoLight',
                fontWeight: FontWeight.w100),
          )
```

And my FontManifest.json:

`[{"family":"MaterialIcons","fonts":[{"asset":"fonts/MaterialIcons-Regular.otf"}]},{"family":"RobotoLight","fonts":[{"asset":"assets/fonts/Roboto/Roboto-Light.ttf", "weight": 100}]},{"family":"packages/cupertino_icons/CupertinoIcons","fonts":[{"asset":"packages/cupertino_icons/assets/CupertinoIcons.ttf"}]}]`

And my pubspec.yaml

```
fonts:
    - family: Roboto
      fonts:
        - asset: assets/fonts/Roboto/Roboto-Medium.ttf
          weight: 500
    - family: RobotoLight
      fonts:
        - asset: assets/fonts/Roboto/Roboto-Light.ttf
          weight: 300
    - family: RobotoThin
      fonts:
        - asset: assets/fonts/Roboto/Roboto-Thin.ttf
          weight: 100
```

**Comment by Justus-M** [comment]
> I can confirm that this is still a problem with rendering smaller font weights in flutter web. I am trying to render a font with weight 300, but it is appearing the same as weight 500. If I run from Android Studio it does work (with html or canvaskit renderers). If I build the project with: `$ flutter build web --web-renderer html` and upload to my web server it DOES NOT work If I build the project with: `$ flutter build web --web-renderer canvaskit` and upload to my web server it DOES work
> 
> So the problem is that the html rendered site does not work. However, following a suggestion from somewhere else, if I create a "dummy" extra font spec in pubspec.yaml just for the weight 300 font (and also matching this in the web > assets > FontManifest.json file) it does seem to work . So changing my pubspec.yaml font spec from:
> 
> ```
> fonts:
>     - family: Roboto
>       fonts:
>         - asset: fonts/Roboto-Light.ttf
>           weight: 300
>         - asset: fonts/Roboto-Medium.ttf
>           weight: 500
> ```
> 
> to
> 
> ```
> fonts:
>     - family: Roboto
>       fonts:
>         - asset: fonts/Roboto-Medium.ttf
>           weight: 500
> 
>     - family: RobotoLight
>       fonts:
>         - asset: fonts/Roboto-Light.ttf
>           weight: 300
> ```
> 
> seems to work. Flutter 3.10.5 stable.

Thanks for sharing! Huge weight off my chest

This works for me, I didn't have to change the FontManifest.json

**Comment by yjbanov** [comment]
We only use a subset of Noto fonts for fallback purposes, so I imagine not all weights are covered. We still need a solution for this. However, according to https://github.com/flutter/flutter/issues/75832#issuecomment-1603019841 there's a workaround.

**Comment by Justus-M** [comment]
> We only use a subset of Noto fonts for fallback purposes, so I imagine not all weights are covered. We still need a solution for this. However, according to [#75832 (comment)](https://github.com/flutter/flutter/issues/75832#issuecomment-1603019841) there's a workaround.

The workaround does not cover all weights.

I used it for Roboto, but w200 is missing. I could not find it anywhere to download.

still much better than having nothing below w400/w500, but it's not a real solution.

**Comment by icecandy** [comment]
@Justus-M that's a strange thing to say:

> 
> The workaround does not cover all weights.
> 
> I used it for Roboto, but w200 is missing. I could not find it anywhere to download.
> 
> still much better than having nothing below w400/w500, but it's not a real solution.

You need the font file to support that font weight. Not all fonts have a full range of weights as it depends if the font designer/font house has developed support for a full set of weights. I'm not sure about Roboto, but quite a lot of fonts do not have the low weights such as 100 and 200. I'm using Poppins which does have a 100 and 200 weight. Actually just looking at my Roboto set, your right, it does not have a 200 weight, but equally well is doesn't have a 900 weight (Black).

Anyway nothing to do with Flutter.

**Comment by Justus-M** [comment]
> @Justus-M that's a strange thing to say:
> 
> > The workaround does not cover all weights.
> > I used it for Roboto, but w200 is missing. I could not find it anywhere to download.
> > still much better than having nothing below w400/w500, but it's not a real solution.
> 
> You need the font file to support that font weight. Not all fonts have a full range of weights as it depends if the font designer/font house has developed support for a full set of weights. I'm not sure about Roboto, but quite a lot of fonts do not have the low weights such as 100 and 200. I'm using Poppins which does have a 100 and 200 weight. Actually just looking at my Roboto set, your right, it does not have a 200 weight, but equally well is doesn't have a 900 weight (Black).
> 
> Anyway nothing to do with Flutter.

Well yes it is to do with flutter and it's the whole point of this thread.

You can use the default flutter text font (roboto) with fontWeight.w200 on all other platforms, but on web nothing below 400 works.

This workaround of specifying the files is not a solution, because you can't recreate fontWeight.w200.

**Comment by icecandy** [comment]
I think you misunderstand how fonts work. You AWAYS need a font file definition. On mobile if you are using a system font then the system gives the OS/app the font specification. If you are using a custom font - that the OS does not provide - then you have to provide the font files to match. In web the fonts are usually libraries that are linked to and downloaded at startup or upon request. In flutter you can use the Google fonts package, but again what that does is download the font file and caches it.

Some frameworks/apps provide a simple "fudge" for creating, for example, a bold version of a font when only the regular version exists. The framework/OS will use the font metric and try to recreate a bold version. But asking it do it for 9 font weights is not practical or possible.

A font file contains the vector description of the font's curves. It is a vector format and so is resolution independent. The framework/OS renders the font as pixels according to it's metrics from the font vector definition ie the font file.

So in summary, once again, you do need the font file specifications and that means for each weight. The later font formats encapsulate all the weights of the font into one single font file, but you still need that single file.

**Comment by Justus-M** [comment]
> I think you misunderstand how fonts work. You AWAYS need a font file definition. On mobile if you are using a system font then the system gives the OS/app the font specification. If you are using a custom font - that the OS does not provide - then you have to provide the font files to match. In web the fonts are usually libraries that are linked to and downloaded at startup or upon request. In flutter you can use the Google fonts package, but again what that does is download the font file and caches it.
> 
> 
> 
> Some frameworks/apps provide a simple "fudge" for creating, for example, a bold version of a font when only the regular version exists. The framework/OS will use the font metric and try to recreate a bold version. But asking it do it for 9 font weights is not practical or possible.
> 
> 
> 
> A font file contains the vector description of the font's curves. It is a vector format and so is resolution independent. The framework/OS renders the font as pixels according to it's metrics from the font vector definition ie the font file.
> 
> 
> 
> So in summary, once again, you do need the font file specifications and that means for each weight. The later font formats encapsulate all the weights of the font into one single font file, but you still need that single file.

You're missing the point.

If a flutter dev builds an iOS, android, Mac, or windows app with the default font and specifies w200 in the TextStyle of a text widget (which does work), they will then have a problem when they build for web (which IS a flutter issue, and the point of this thread).

Then maybe they come to this thread and see the "solution" for their problem, but actually it's not a solution because like you said, Roboto font files don't include w200.

**Comment by jynd** [comment]
> I can confirm that this is still a problem with rendering smaller font weights in flutter web. I am trying to render a font with weight 300, but it is appearing the same as weight 500. If I run from Android Studio it does work (with html or canvaskit renderers). If I build the project with: `$ flutter build web --web-renderer html` and upload to my web server it DOES NOT work If I build the project with: `$ flutter build web --web-renderer canvaskit` and upload to my web server it DOES work
> 
> So the problem is that the html rendered site does not work. However, following a suggestion from somewhere else, if I create a "dummy" extra font spec in pubspec.yaml just for the weight 300 font (and also matching this in the web > assets > FontManifest.json file) it does seem to work . So changing my pubspec.yaml font spec from:
> 
> ```
> fonts:
>     - family: Roboto
>       fonts:
>         - asset: fonts/Roboto-Light.ttf
>           weight: 300
>         - asset: fonts/Roboto-Medium.ttf
>           weight: 500
> ```
> 
> to
> 
> ```
> fonts:
>     - family: Roboto
>       fonts:
>         - asset: fonts/Roboto-Medium.ttf
>           weight: 500
> 
>     - family: RobotoLight
>       fonts:
>         - asset: fonts/Roboto-Light.ttf
>           weight: 300
> ```
> 
> seems to work. Flutter 3.10.5 stable.

yes, it's work for me. Thanks. everybody can try it
